src/Coordination/KeeperConnectionStats.cpp

```diff
@@ -1,65 +1,63 @@
+#include <atomic>
 #include <Coordination/KeeperConnectionStats.h>
 
 namespace DB
 {
 
 uint64_t KeeperConnectionStats::getMinLatency() const
 {
-    return min_latency;
+    return min_latency.load(std::memory_order_relaxed);
 }
 
 uint64_t KeeperConnectionStats::getMaxLatency() const
 {
-    return max_latency;
+    return max_latency.load(std::memory_order_relaxed);
 }
 
 uint64_t KeeperConnectionStats::getAvgLatency() const
 {
-    if (count != 0)
-        return total_latency / count;
+    auto cnt = count.load(std::memory_order_relaxed);
+    if (cnt)
+        return total_latency.load(std::memory_order_relaxed) / cnt;
     return 0;
 }
 
 uint64_t KeeperConnectionStats::getLastLatency() const
 {
-    return last_latency;
+    return last_latency.load(std::memory_order_relaxed);
 }
 
 uint64_t KeeperConnectionStats::getPacketsReceived() const
 {
-    return packets_received;
+    return packets_received.load(std::memory_order_relaxed);
 }
 
 uint64_t KeeperConnectionStats::getPacketsSent() const
 {
-    return packets_sent;
+    return packets_sent.load(std::memory_order_relaxed);
 }
 
 void KeeperConnectionStats::incrementPacketsReceived()
 {
-    packets_received++;
+    packets_received.fetch_add(1, std::memory_order_relaxed);
 }
 
 void KeeperConnectionStats::incrementPacketsSent()
 {
-    packets_sent++;
+    packets_sent.fetch_add(1, std::memory_order_relaxed);
 }
 
 void KeeperConnectionStats::updateLatency(uint64_t latency_ms)
 {
-    last_latency = latency_ms;
-    total_latency += (latency_ms);
-    count++;
-
-    if (latency_ms < min_latency)
-    {
-        min_latency = latency_ms;
-    }
-
-    if (latency_ms > max_latency)
-    {
-        max_latency = latency_ms;
-    }
+    last_latency.store(latency_ms, std::memory_order_relaxed);
+    total_latency.fetch_add(latency_ms, std::memory_order_relaxed);
+    count.fetch_add(1, std::memory_order_relaxed);
+
+    if (latency_ms < min_latency.load(std::memory_order_relaxed))
```

Comment:
This is not correct (can lose data). Atomic min/max requires CAS.

Reply:
Yes, I am lazy. Fix now.